src/components/Forms/TextField.js

```diff
@@ -0,0 +1,268 @@
+    WebkitTapHighlightColor: 'rgba(0,0,0,0)'
+  },
+  highlight: {
+    border: `1px solid ${colors.GREEN_500}`
+  },
+  inputDisabled: {
+    border: `1px dashed ${colors.GRAY_74}`,
+    backgroundColor: colors.GRAY_93,
+    color: colors.GRAY_46,
+    cursor: 'not-allowed'
+  },
+  inputError: {
+    border: `1px solid ${colors.RED_700}`,
+    backgroundColor: '#FDE6EB'
+  },
+  fullWidth: {
+    width: '100%'
+  },
+  halfWidth: {
+    width: '162px'
+  }
+}
+
+@FormComponent
+@Radium
+class TextField extends React.Component {
+  static propTypes = {
+    /** Name of the field */
+    name               : PropTypes.string.isRequired,
+    /** HTML autocomplete attribute */
+    autoComplete       : PropTypes.string,
+    /** DefaultValue for non controlled component */
+    defaultValue       : PropTypes.any,
+    /** Disable the text field */
+    disabled           : PropTypes.bool,
+    /** Text of label that will animate when TextField is focused */
+    floatingLabelText  : PropTypes.string,
+    /** Sets width to 100% */
+    fullWidth          : PropTypes.bool,
+    /** Sets width to 162px */
+    halfWidth          : PropTypes.bool,
+    /** Helper text will show up in bottom right corner below TextField */
+    helperText         : PropTypes.string,
+    /** Hint text will show up when input is focused and there is no value */
+    hintText           : PropTypes.string,
+    /** Style for hint text */
+    hintTextStyle      : PropTypes.object,
+    /** Uniq id for input */
+    id                 : PropTypes.string,
+    /** Style for input */
+    inputStyle         : PropTypes.object,
+    /** Set by FormComponent by default.   */
+    isValid            : PropTypes.bool,
+    /** onFocus callback */
+    onFocus            : PropTypes.func,
+    /** onChange callback */
+    onChange           : PropTypes.func,
+    /** onBlur callback */
+    onBlur             : PropTypes.func,
+    /** onKeyDown callback */
+    onKeyDown          : PropTypes.func,
+    /** Mark the field as required.  */
+    required           : PropTypes.bool,
+    /** Error from server to show ServerError message */
+    serverError        : PropTypes.string,
+    /** Wrapper styles */
+    style              : PropTypes.object,
+    /** Input type ie. 'text', 'email', password, etc..  */
+    type               : PropTypes.string,
+    /** Text to show for validation error  */
+    validationErrorText: PropTypes.string,
+    /** Value will make TextField a controlled component  */
+    value              : PropTypes.string,
+  }
+
+  static defaultProps = {
+    autoComplete: 'on',
+    disabled    : false,
+    type        : 'text',
+    defaultValue: null
+  }
+
+  state = {
+    hasValue: this.props.defaultValue !== null || Boolean(this.props.value)
+  }
+
+  componentWillMount() {
+    const { name } = this.props
+
+    const uniqueId = `${name}-${Math.floor(Math.random() * 0xFFFF)}`
+    this.uniqueId = uniqueId.replace(/[^A-Za-z0-9-]/gi, '')
```

Comment:
let's add a comment here for why we're generating this unique id. It also looks possible that the implementer can pass in their own id as a prop, which is what gets used first. Maybe we should make this a default prop rather than always run on mount? That way the docs would also mark it as optional with one generated for you.

Reply:
Only issue with making it a default prop is that we don't have access to `this.props.name` in defaultProps

Reply:
Yeah, that's true. In that case maybe just a quick check for it in the prop during mount

```js
this.uniqueId = this.props.id || <generate one>
```